### PR TITLE
Fix lgtm issues

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

- Add some null checks.
- Remove unused code.

## PR Context  

Come from https://lgtm.com/projects/g/PowerShell/PowerShell/alerts/?mode=tree&ruleFocus=1506101336231

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by RDIL** [comment]
It seems like this PR is making the Code factor grade decline.

**Comment by iSazonov** [comment]
@RDIL No, the issues already was there. CodeFactor has strange reports sometimes.

**Comment by RDIL** [comment]
Ah okay.

**Review by markekraus — APPROVED** [review]
Web cmdlet changes  LGTM

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs [hunk 1]
```diff
@@ -1451,7 +1451,7 @@ public class DriveMatchingCoreCommandBase : CoreCommandBase
                     }
                     else
                     {
-                        if (nameMatcher.IsMatch(drive.Name))
+                        if (nameMatcher != null && nameMatcher.IsMatch(drive.Name))
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs]: Can this be 
```suggestion
                        if (nameMatcher?.IsMatch(drive.Name))
```

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 2]
```diff
@@ -8071,8 +8071,10 @@ private static string WinInternalGetTarget(SafeFileHandle handle)
                     targetDir = Encoding.Unicode.GetString(reparseDataBufferMountPoint.PathBuffer, reparseDataBufferMountPoint.SubstituteNameOffset, reparseDataBufferMountPoint.SubstituteNameLength);
                 }
 
-                if (targetDir.StartsWith(NonInterpretedPathPrefix, StringComparison.OrdinalIgnoreCase))
+                if (targetDir != null && targetDir.StartsWith(NonInterpretedPathPrefix, StringComparison.OrdinalIgnoreCase))
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Can this be
```suggestion
                if (targetDir?.StartsWith(NonInterpretedPathPrefix, StringComparison.OrdinalIgnoreCase))
```

**Review by adityapatwardhan — APPROVED** [review]
Approved with non-blocking suggestions

**Review by vexx32 — COMMENTED** [review]
Love the cleanup! 👍 😍

**Review by RDIL — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs]: Report is "Cannot implicitly convert type bool? to bool."

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 2 above)
Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: The same.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 3]
```diff
@@ -1193,29 +1193,24 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
...
                 }
 
-                if (content is FormObject)
+                if (content is FormObject form)
                 {
-                    FormObject form = content as FormObject;
                     SetRequestContent(request, form.Fields);
                 }
-                else if (content is IDictionary && request.Method != HttpMethod.Get)
+                else if (content is IDictionary dictionary && request.Method != HttpMethod.Get)
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Purpose of this PR is to fix null checks. So code refactoring is out of the PR. Fill free to push new PR if you see benefits.

File: src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs [hunk 4]
```diff
@@ -692,21 +692,19 @@ private void ProcessCachedGroupOnTable(TableHeaderInfo thi, List<PacketInfoData>
...
-                        widths[kk] = cellCount;
+                    TableRowEntry tre = fed.formatEntryInfo as TableRowEntry;
+                    int kk = 0;
 
-                    kk++;
+                    foreach (FormatPropertyField fpf in tre.formatPropertyFieldList)
+                    {
+                        cellCount = _lo.DisplayCells.Length(fpf.propertyValue);
+                        if (widths[kk] < cellCount)
+                            widths[kk] = cellCount;
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs]: The same.

File: src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs [hunk 5]
```diff
@@ -729,19 +727,17 @@ private void ProcessCachedGroupOnWide(WideViewHeaderInfo wvhi, List<PacketInfoDa
...
-                    if (cellCount > maxLen)
-                        maxLen = cellCount;
+                    WideViewEntry wve = fed.formatEntryInfo as WideViewEntry;
+                    FormatPropertyField fpf = wve.formatPropertyField as FormatPropertyField;
+
+                    if (!string.IsNullOrEmpty(fpf.propertyValue))
+                    {
+                        cellCount = _lo.DisplayCells.Length(fpf.propertyValue);
+                        if (cellCount > maxLen)
+                            maxLen = cellCount;
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs]: The same.

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs]: @vexx32 I consider such patterns as a "security" risk.

[the remark this answers is not in the document]
Comment (vexx32) [reply src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs]: Could you elaborate @iSazonov?

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs]: https://blog.codecentric.de/en/2014/02/curly-braces/

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs (hunk 1 above)
Comment (adityapatwardhan) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs]: Maybe this works?
```CSharp
if (nameMatcher?.IsMatch(drive.Name) ?? false)
{
}
```

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: same as above

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs]: Looks as puzzle :-)

**Pull request merged**